src/test/java/com/microsoft/sqlserver/jdbc/bulkCopy/ImpISQLServerBulkRecord_IssuesTest.java

```diff
@@ -0,0 +1,435 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.bulkCopy;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import java.io.IOException;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.ISQLServerBulkRecord;
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCopy;
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+import com.microsoft.sqlserver.testframework.Utils;
+
+@RunWith(JUnitPlatform.class)
+public class ImpISQLServerBulkRecord_IssuesTest extends AbstractTest {
+
+    static Statement stmt = null;
+    static PreparedStatement pStmt = null;
+    static String query;
+    static SQLServerConnection con = null;
+    static String srcTable = "sourceTable";
+    static String destTable = "destTable";
+    String variation;
+
+    /**
+     * Testing that sending a bigger varchar(3) to varchar(2) is thowing the proper error message.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testVarchar() throws Exception {
+        variation = "testVarchar";
+        BulkDat bData = new BulkDat(variation);
+        String value = "aa";
+        query = "CREATE TABLE " + destTable + " (smallDATA varchar(2))";
+        stmt.executeUpdate(query);
+
+        try {
+            SQLServerBulkCopy bcOperation = new SQLServerBulkCopy(connectionString);
+            bcOperation.setDestinationTableName(destTable);
+            bcOperation.writeToServer(bData);
+            bcOperation.close();
+        }
+        catch (Exception e) {
+            if (e instanceof SQLServerException) {
+                assertTrue(e.getMessage().contains("The given value of type"), "Invalid Error message: " + e.toString());
+            }
+        }
+        ResultSet rs = stmt.executeQuery("select * from " + destTable);
```

Comment:
this returns an empty resultset since nothing was inserted. No point in the while loop